DataStructure/LinkList/linklist.java

```diff
@@ -0,0 +1,455 @@
+		}
+
+	}
+
+	private Node head = null;
+	private Node tail = null;
+
+	// Insertion in List
+	void insertiontolast(int data) {
+		if (head == null) {
+			Node temp = new Node(data);
+			head = temp;
+			tail = temp;
+			return;
+		} else {
+
+			Node temp = new Node(data);
+			tail.next = temp;
+			tail = tail.next;
+
+		}
+
+	}
+
+	// insertion from head
+
+	void insertiontohead(int data) {
+		if (head == null) {
+			Node temp = new Node(data);
+			head = temp;
+			tail = temp;
+		} else {
+			Node temp = new Node(data);
+			temp.next = head;
+			head = temp;
+
+		}
+
+	}
+
+	// insertion from Any Position In Linklist
+
+	void insertionatanypos(int data, int pos) {
+
+		if (pos < 1) {
+			System.out.println("Invalid Postion");
+			return;
+		}
+		if (pos == 1) {
+			insertiontohead(data);
+			return;
+		} else {
+			int count = 1;
+			Node temp = head;
+			while (count < pos - 1 && temp != null) {
+				temp = temp.next;
+				count++;
+			}
+			Node newnode = new Node(data);
+			if (count == pos - 1 && temp.next == null) {
+				insertiontolast(data);
+				return;
+
+			} else if (temp == null && count <= pos) {
+				System.out.println("Invalid Position");
+				return;
+
+			} else if (temp != null) {
+				newnode.next = temp.next;
+				temp.next = newnode;
+
+			}
+
+		}
+
+	}
+
+	// deletion from head
+
+	int deletefromhead() {
+		if (head == null) {
+			return -1;
+
+		}
+		if (head.next == null) {
+			int temp = head.data;
+			head = tail = null;
+			return temp;
+		} else {
+			Node temp = head;
+			head = head.next;
+			return temp.data;
+		}
+	}
+
+	int deletefromlast() {
+		if (head == null) {
+
+			return -1;
+
+		}
+		if (head.next == null) {
+			int temp = head.data;
+			head = tail = null;
+			return temp;
+		}
+
+		Node temp = head;
+		Node temp2 = head.next;
+		while (temp2.next != null) {
+			temp = temp2;
+			temp2 = temp2.next;
+		}
+		tail = temp;
+		tail.next = null;
+		return temp2.data;
+	}
+
+	// deletion from Any Position
+
+	int deletefromanypos(int pos) {
+		if (isempty()) {
+			return -1;
+		}
+		if (pos < 1) {
+			return -1;
+		}
+		if (pos == 1) {
+			return deletefromhead();
+		} else {
+			int count = 1;
+			Node temp = head;
+			Node temp1 = null;
+			while (count < pos && temp != null) {
+				temp1 = temp;
+				temp = temp.next;
+				count++;
+			}
+			if (temp == null && count <= pos) {
+				return -1;
+			}
+
+			temp1.next = temp.next;
+			return temp.data;
+
+		}
+
+	}
+
+	// Sort List
+	void sort() {
+		if (head == null) {
+			System.out.println("-1");
+			return;
+		}
+
+		head = mergesort(head);
+
+	}
+
+	// Sort List using Mergesort
+	private Node mergesort(Node head) {
+		if (head.next == null) {
+			return head;
+		}
+
+		Node mid = midelement(head);
+		Node temp2 = mid.next;
+		mid.next = null;
+
+		Node first = mergesort(head);
+		Node second = mergesort(temp2);
+
+		Node temp = merge(first, second);
+
+		return temp;
+	}
+
+	// Find the middle element in list
+	private Node midelement(Node temp) {
```

Comment:
Proper Camel Casing should be used while naming variables like - midelement should be changed with midElement. Change this throughout the modules.